### PR TITLE
Fixes bug 1493041 - Topstories feed uninit can remove idle-daily obse…

### PR DESCRIPTION
**Pull request by ScottDowne**
Steps to test here: https://bugzilla.mozilla.org/show_bug.cgi?id=1493041

### CODE REVIEW

File: test/unit/lib/TopStoriesFeed.test.js [hunk 1]
```diff
@@ -1021,6 +1021,18 @@ describe("Top Stories Feed", () => {
...
+      let callOrder = [];
+      instance.addObserver = () => {
+        callOrder.push("add");
+      };
+      instance.removeObserver = () => {
+        callOrder.push("remove");
+      };
+      await Promise.all([instance.onInit(), instance.uninit()]);
+      assert.equal(callOrder[0], "add");
+      assert.equal(callOrder[1], "remove");
```
Comment (ScottDowne) [inline test/unit/lib/TopStoriesFeed.test.js]: Dunno a better way to test the call order, and in general this was one of the more tricky tests to write, and while the above seems to get the job done, happy to take improvements.

File: lib/TopStoriesFeed.jsm [hunk 2]
```diff
@@ -102,10 +101,18 @@ this.TopStoriesFeed = class TopStoriesFeed {
...
-    Services.obs.removeObserver(this, "idle-daily");
+    this.removeObserver("idle-daily");
     SectionsManager.disableSection(SECTION_ID);
   }
 
+  addObserver(observer) {
+    Services.obs.addObserver(this, observer);
+  }
+
+  removeObserver(observer) {
```
Comment (ScottDowne) [inline lib/TopStoriesFeed.jsm]: I added these to help me test...

**Comment by ScottDowne** [comment]
Steps to test here: https://bugzilla.mozilla.org/show_bug.cgi?id=1493041

File: test/unit/lib/TopStoriesFeed.test.js (hunk 1 above)
Comment (k88hudson) [reply test/unit/lib/TopStoriesFeed.test.js]: Could you do something like 
```
 const initPromise = init();
  assert.calledOnce(Services.obs.addObserver);
  await initPromise;
```

**Comment by ScottDowne** [comment]
@k88hudson thanks for the suggestions, much better. Updated.

**Pull request merged**